caikit/runtime/http_server/http_server.py

```diff
@@ -635,6 +635,7 @@ async def _handler(
                         output_streaming=False,
                         task=rpc.task,
                         aborter=aborter,
+                        context=context,
```

Comment:
The main question I have on this PR is whether we need a `caikit.core` abstract definition of a `Context` so that modules have a contractual API signature when choosing to query the context. It would mean implementing some kind of common shim between HTTP and gRPC for common operations. This would also be particularly useful for tracing so that once we've got [trace instrumentation](https://github.com/caikit/caikit/pull/719) in place, the body of a module could fetch a request ID and forward it to downstream calls.